iotiletest/iotile/fixture/device_fixture.py

```diff
@@ -23,8 +24,12 @@ def pytest_generate_tests(metafunc):
         ids = [int(x, 0) for x in metafunc.config.option.tile]
         metafunc.parametrize("tile", ids, scope='session')
 
+    if 'direct' in metafunc.fixturenames:
+        connections = [x for x in metafunc.config.option.device]
+        metafunc.parametrize("direct", connections, scope='session')
+
 @pytest.fixture(scope='module')
-def device(port, device_id):
+def device(port, device_id, direct):
```

Comment:
For backwards compatibility of the function signature, this should have a default value of `direct=False`.  I believe this would break otherwise if the user didn't pass `--direct` on the command line.

Reply:
I have tried `direct=False` and `direct=[]` and in both cases the default value gets used rather than the overridden value. I'm not sure exactly what is going on. I did add a default in the option parser though so that should take care of the default case.